Makefile

```diff
@@ -25,7 +25,7 @@ NOTIFY_CREDENTIALS ?= ~/.notify-credentials
 bootstrap: generate-version-file ## Set up everything to run the app
 	pip3 install -r requirements_for_test.txt
 	createdb notification_api || true
-	. environment.sh && flask db upgrade
+	. environment.sh || flask db upgrade
```

Comment:
Would be interested to know your choice of bash syntax here (partly for my own learning).

So this changes it from an `and` to an `or`. Do we want an `or` here? It means that if there was an environment.sh file in someones dev environment but their db upgrade failed then `make bootstrap` would still appear successful?

Could another option be 

```
createdb notification_api || true	
. environment.sh || true	
flask db upgrade
```

I write bash fairly rarely so feel free to correct me with any incorrect statements/assumptions I may have made

In principle, I think the approach of trying to source the environment.sh file and it not succeeding but the process continuing on is acceptable

Reply:
Sure, I can explain more:

- First thing to note is that each line of the make rule is run in isolation, so doing `. environment.sh` on one line and `flask db upgrade` on another line won't work.

- You're right that this change means `. environment.sh` will fail silently if someone doesn't have the file. However, `flask db upgrade` _will_ then fail loudly.

I think this is OK because:

- Frankly, people should read the README, which tells them to create an `environment.sh` file 😉.
- The error from `flask db upgrade` is pretty good indicator that the `environment.sh` file is missing.

Does that help?

Reply:
Thanks, just to further clarify

> First thing to note is that each line of the make rule is run in isolation, so doing . environment.sh on one line and flask db upgrade on another line won't work.

Is the fact each rule is run in isolation because it is bash or is this because it is a make file?
And have you tested this assumption or got a link to back it up out of interest? If not I can run a quick test to confirm it for myself

Another option to ponder would be

```
(environment.sh || true) &&  flask db upgrade
```

Not sure if that is valid bash with the brackets but then it is essentially optional that environment.sh can be sourced but required that the db can be upgraded

Reply:
> Is the fact each rule is run in isolation because it is bash or is this because it is a make file?

It's because it's a Makefile. I think I learnt that through trial and error, so it's just background knowledge for me now. You can find accounts of it online too (e.g. [here](https://stackoverflow.com/questions/54481072/source-env-variables-files-inside-makefile)).

The default shell for `make` is (apparently) `/bin/sh` by the way.

> `(environment.sh || true) &&  flask db upgrade`

Oops. This led me to remember `||` is lazy, so `.environment.sh || flask db upgrade` won't get to `flask db upgrade` 😬. I've amended the commit with something that actually works.

Thanks for persisting with the discussion ⭐. Do you want to take another look?